### PR TITLE
ui: add slack user groups to on call notifications

### LINKED ISSUE #1228: Update slack group users to match the current on-call rotation.
**Is your feature request related to a problem? Please describe:**
Not a problem, however it would be nice if we could allow go-alert to manage the users in our support slack group, that way we could automatically match go-alert's on call rotation with who is assigned to the slack group.

**Describe the solution you'd like:**
When a go-alert rotation ends:
- go alert would add the new on-call users (primary and secondary rotations) to a configured slack group
- go alert would remove the old on-call users (primary and secordary rotations) from the same configured slack group

**Describe alternatives you've considered:**
Currently our team does this manually, however because our goalert rotation ends at midnight thursday evening. Because of this we tend to either update the slack group with the new on call rotation either the day before the rotation ends, or the morning after, potentially sending some of our slack based notifications to the wrong on call group (i.e. if someone asks for help from our `@<team-name>-support` group in slack.

**Additional context:**
N/A

Thanks so much!

### PR DESCRIPTION
**Pull request by Forfold**
**Description:**
This PR introduces the ability to create an on-call notification rule that simultaneously keeps a slack user group in sync with the list of GoAlert on-call users

**Which issue(s) this PR fixes:**
Part of #1228
EDIT: not marking as closed until feature flag is removed

**Screenshots:**
Screenshot 2023-04-11 at 3 29 13 PM

Screenshot 2023-04-11 at 3 29 18 PM

Screenshot 2023-04-11 at 3 29 07 PM

### CODE REVIEW

**Comment by Forfold** [comment]
Currently includes changes from #2926. Let's get that merged first.

File: notification/slack/usergrouperror.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	for _, u := range e.Missing {
+		urlStr := e.callbackFunc(fmt.Sprintf("users/%s", url.PathEscape(u.ID)))
+		refs = append(refs, slackLink(urlStr, u.Name))
+	}
+	return strings.Join(refs, ", ")
+}
+
+// ScheduleRef returns a string that can be used to reference the schedule in a Slack message.
+func (e userGroupError) ScheduleRef() string {
+	urlStr := e.callbackFunc("schedules/" + url.PathEscape(e.ScheduleID))
```
Comment (AlaricWhitney) [inline notification/slack/usergrouperror.go:49]: To promote good secure coding habits, we should never directly concatenate strings.
```suggestion
	urlStr := e.callbackFunc(fmt.Sprintf("schedules/%s", url.PathEscape(e.ScheduleID)))
```

File: notification/slack/usergroupsender.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		return nil, fmt.Errorf("lookup team ID: %w", err)
+	}
+
+	var userIDs []string
+	for _, u := range t.Users {
+		userIDs = append(userIDs, u.ID)
+	}
+
+	userSlackIDs := make(map[string]string, len(t.Users))
+	err = s.cfg.UserStore.AuthSubjectsFunc(ctx, "slack:"+teamID, userIDs, func(sub user.AuthSubject) error {
```
Comment (AlaricWhitney) [inline notification/slack/usergroupsender.go:58]: To promote good secure coding habits, we should never directly concatenate strings.
```suggestion
	err = s.cfg.UserStore.AuthSubjectsFunc(ctx, fmt.Sprintf("slack:%s", teamID), userIDs, func(sub user.AuthSubject) error {
```

**Review by AlaricWhitney — CHANGES_REQUESTED** [review]

**Review by AlaricWhitney — APPROVED** [review]
LGTM

**Comment by mastercactapus** [comment]
Switching this to draft while blocked, #2926 has API changes that need to be settled first.

**Review by mastercactapus — CHANGES_REQUESTED** [review]
This is still visible without the feature flag set, let's add a guard around that. EP step UI has an example of how to do this for webhooks.

Also, couple of layout questions:
- Should we move the input fields below the notification option? I feel the input should follow the context. Also, we may be adding webhook support here soon, so it will help prepare for more noise.
- Can we put a checkbox in for the user group and conditionally show the input field? I think that might reduce noise in the form while still keeping it clear how to opt-in to the new feature.

**Review by mastercactapus — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -182,7 +182,6 @@ ensure-yarn: # Yarn ensures the correct version of yarn is installed
 yarn:
 	corepack enable
 	corepack prepare yarn@stable --activate
-	yarn set version stable
```
Comment (mastercactapus) [inline Makefile:185]: This is no longer needed with #2984

**Review by AlaricWhitney — APPROVED** [review]

**Pull request merged**